sources/Simulink/src/BlockFactory.cpp

```diff
@@ -836,8 +839,11 @@ static void mdlRTW(SimStruct* S)
             return;
         }
 
+        // Get the block object name
+        std::string blockName(block->getName(blockInfo));
```

Comment:
Can we use `blockInfo->getName(...)` directly here, so we can avoid adding the `getName` in the `Block` class?

Reply:
Or perhaps there is some additional advantage in adding `Block::getName` that I am missing?

Reply:
I was not really concern by the number of  function calls to be honest, but rather on the logical division of the methods between `Block` or `BlockInformation` : for me as this name is the unique name of the instance, it is more clean to have this name exposed just in BlockInformation, instead of "polluting" the `Block` interface with methods that do add a lot. 
However, I do not have a strong opinion on this, feel free to leave the `Block::getName`.

Reply:
Something similar to `Block::getUniqueName` happens to `Block::getParameters`.

Since both of them are not part of the interface (they are not virtual) and these methods are never called in the plugins, I would keep them there for the time being and postpone the discussion of moving both to the `BlockInformation` class.

Related to this, sooner or later we should also clean a little all the parameters logic. It was quite difficult to make it work in the first place and it needs a polished.